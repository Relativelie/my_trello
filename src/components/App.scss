@import "../templates";


.listsBlock {
    display: flex;
    height: calc(100vh - ($addListMargin + $addListHeight));
}