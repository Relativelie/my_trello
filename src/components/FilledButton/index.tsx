import './index.scss';

type FilledButtonProps = {
  onClick: () => void;
  children: React.ReactNode;
};

const FilledButton: React.FC<FilledButtonProps> = ({ onClick, children }) => {
  return (
    <button className="filled-button" onClick={onClick} type="button">
      {children}
    </button>
  );
};

export default FilledButton;
