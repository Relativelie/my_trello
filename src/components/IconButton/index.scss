@import '../../styles/templates.scss';

.icon-button {
  padding: 1px;
}

.deleting-icon-button {
  @include optionBtn(
    $bgColor: rgba(253, 161, 161, 0.624),
    $bgColorHover: rgb(253, 52, 52),
    $isIconBtn: true
  );
}

.common-icon-button {
  @include optionBtn(
    $bgColor: rgba(205, 205, 205, 0.793),
    $bgColorHover: rgb(80, 168, 246),
    $isIconBtn: true
  );
}

.icon-button_icon {
  position: absolute;
  top: 0;
  left: 0;

  & svg,
  img {
    width: 20px;
    height: 20px;
  }
}
