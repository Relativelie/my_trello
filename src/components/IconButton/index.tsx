import { ButtonVariantsENUM, buttonStyles } from './models';
import './index.scss';

type AppIconButtonProps = {
  children: React.ReactNode;
  onClick?: () => void;
  variant?: ButtonVariantsENUM;
};

const AppIconButton: React.FC<AppIconButtonProps> = ({
  onClick,
  children,
  variant = ButtonVariantsENUM.common,
}) => {
  return (
    <button className={buttonStyles[variant]} onClick={onClick} type="button">
      <div className="icon-button_icon icon-button">{children}</div>
    </button>
  );
};

export default AppIconButton;
