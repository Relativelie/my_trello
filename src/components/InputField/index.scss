@import "../../templates";

.inputContainer {
  grid-area: name;
}

.inputContainer__inputName {
  width: 95%;
  cursor: pointer;
}

.inputContainer__inputName_hideInput {
  opacity: 0;
}
