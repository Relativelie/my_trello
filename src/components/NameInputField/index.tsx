import { SyntheticEvent, FocusEvent } from 'react';

import { useActions } from '../../hooks/useActions';
import { useTypedSelector } from '../../hooks/useTypedSelector';

import './index.scss';

interface NameInputFieldProps {
  index: number[] | number;
  typeOfElement: string;
  value: string;
}

const NameInputField: React.FC<NameInputFieldProps> = ({
  index,
  typeOfElement,
  value,
}) => {
  const { nameValidationOn, showCurrentValueInInput } = useActions();
  const { lists } = useTypedSelector((state) => state.listOfTasksReducer);
  const { currentName } = useTypedSelector(
    (inputFieldState) => inputFieldState.inputFieldReducer,
  );

  const validateInputValue = (
    eventType: string,
    blurEvent: SyntheticEvent<EventTarget>,
  ) => {
    if (eventType === 'blur' || eventType === 'Enter') {
      const elem = blurEvent.target as HTMLInputElement;
      const inputValue = elem.defaultValue;
      const indexElem = elem?.dataset?.elemkey?.split(',');
      if (indexElem !== undefined) {
        nameValidationOn(inputValue, indexElem, typeOfElement);
      }
      elem.classList.add('inputContainer__inputName_hideInput');
      elem.blur();
    }
  };

  const showInput = (e: FocusEvent<HTMLInputElement>) => {
    let inputValue;
    if (typeof index === 'number') {
      inputValue = lists[index];
    } else inputValue = value;
    showCurrentValueInInput(inputValue);
    e.target.classList.remove('inputContainer__inputName_hideInput');
  };

  return (
    <div className="inputContainer">
      <input
        aria-label={`add new ${typeOfElement}`}
        data-elemkey={index}
        className="inputContainer__inputName inputContainer__inputName_hideInput"
        type="text"
        maxLength={44}
        onKeyPress={(blurEvent) => {
          validateInputValue(blurEvent.key, blurEvent);
        }}
        onBlur={(blurEvent) => {
          validateInputValue(blurEvent.type, blurEvent);
        }}
        value={currentName || ''}
        onChange={(e) => showCurrentValueInInput(e.target.value)}
        onFocus={(e) => showInput(e)}
      />
    </div>
  );
};

export default NameInputField;
