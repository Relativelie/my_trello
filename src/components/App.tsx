import { Children } from 'react';
import { DragDropContext, Droppable, DropResult } from 'react-beautiful-dnd';

import { useTypedSelector } from '../hooks/useTypedSelector';
import { useActions } from '../hooks/useActions';

import { AddList } from './addList/AddList';
import { Lists } from './lists/Lists';
import { Tasks } from './tasks/Tasks';
import { Menu } from './menu/Menu';

import './App.scss';

export default function App() {
    const { dragDropList,
        dragDropListWithTasks,
        dragDropTasks,
        addNewList,
        addTaskArrayToNewList } = useActions();
    const { lists } = useTypedSelector((state) => state.listOfTasksReducer);
    const { tasks } = useTypedSelector((tasksState) => tasksState.tasksReducer);

    const handleOnDragEnd = (e: DropResult) => {
        if (e.destination !== undefined && e.destination !== null) {
            const indexTo = e.destination.index;
            const indexFrom = e.source.index;
            if (e.type === 'lists') {
                dragDropList(indexTo, indexFrom);
                dragDropListWithTasks(indexTo, indexFrom);
            } else {
                const listFrom = getValueFromStr(e.source.droppableId, 9);
                const listTo = getValueFromStr(e.destination.droppableId, 9);
                dragDropTasks(listTo, listFrom, indexTo, indexFrom);
            }
        }
    };

    const getValueFromStr = (value: string, countOfStr: number) => {
        return parseInt(value.substring(value.length, countOfStr), 10);
    };

    const listsAdding = () => {
        addNewList();
        addTaskArrayToNewList(lists.length);
    };

    return (
        <div>
            <Menu />
            <main>
                <AddList listsAdding={listsAdding} />
                <DragDropContext onDragEnd={(e) => handleOnDragEnd(e)}>
                    <Droppable droppableId="lists" type="lists" direction="horizontal">
                        {(provided) => (
                            <div className="listsBlock" {...provided.droppableProps} ref={provided.innerRef}>
                                <div className="listsBlock_container">
                                    {lists.map((list: string, listIndex: number) => (
                                        <Lists key={listIndex} list={list} index={listIndex}>
                                            {Children.map(tasks[listIndex], (child, taskIndex) => (
                                                <Tasks
                                                    listIndex={listIndex}
                                                    taskIndex={taskIndex}
                                                    task={child}
                                                />
                                            ))}
                                        </Lists>
                                    ))}
                                </div>
                                {provided.placeholder}
                            </div>
                        )}
                    </Droppable>
                </DragDropContext>
            </main>
        </div>
    );
}
