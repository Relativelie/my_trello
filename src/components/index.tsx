import FilledButton from './FilledButton';
import AppIconButton from './IconButton';
import InputField from './InputField';
import Menu from './Menu';
import OutlinedButton from './OutlinedButton';

export { OutlinedButton, Menu, InputField, AppIconButton, FilledButton };
