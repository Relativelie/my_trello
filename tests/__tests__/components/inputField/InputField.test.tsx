import '@testing-library/jest-dom/extend-expect';
import userEvent from '@testing-library/user-event';
import { fireEvent, screen } from '@testing-library/dom';
import { NameInputField } from '../../../../src/components/NameInputField';
import { render } from '../../../testsSetup/test-utils';

describe('Input field component', () => {
  test('snapshot - List Input field component', () => {
    const tree = render(
      <NameInputField index={0} typeOfElement="list" taskValue="" />,
    );
    expect(tree).toMatchSnapshot();
  });

  test('snapshot - Task Input field component', () => {
    const tree = render(
      <NameInputField index={[0, 0]} typeOfElement="task" taskValue="0 123" />,
    );
    expect(tree).toMatchSnapshot();
  });

  test('List input - show and hide(with enter) input field', () => {
    render(<NameInputField index={0} typeOfElement="list" taskValue="" />);
    const inputEl = screen.getByLabelText(/add new list/i);
    expect(inputEl).toHaveClass(
      'inputContainer__inputName inputContainer__inputName_hideInput',
    );
    userEvent.click(inputEl);
    expect(inputEl).toHaveClass('inputContainer__inputName');
    fireEvent.keyPress(inputEl, { key: 'Enter', keyCode: 13 });
    expect(inputEl).toHaveClass(
      'inputContainer__inputName inputContainer__inputName_hideInput',
    );
  });

  test('List input - show and hide(with tab) input field', () => {
    render(<NameInputField index={0} typeOfElement="list" taskValue="" />);
    const inputEl = screen.getByLabelText(/add new list/i);
    expect(inputEl).toHaveClass(
      'inputContainer__inputName inputContainer__inputName_hideInput',
    );
    userEvent.click(inputEl);
    expect(inputEl).toHaveClass('inputContainer__inputName');
    userEvent.tab();
    expect(inputEl).toHaveClass(
      'inputContainer__inputName inputContainer__inputName_hideInput',
    );
  });

  test('List input - input some value', () => {
    render(<NameInputField index={0} typeOfElement="list" taskValue="" />);
    const inputEl = screen.getByLabelText(/add new list/i);
    userEvent.click(inputEl);
    userEvent.type(inputEl, 'some text for input 123!@#$%^&*()_+');
    expect(inputEl).toHaveValue('some text for input 123!@#$%^&*()_+');
    userEvent.tab();
    expect(inputEl).toHaveValue('some text for input 123!@#$%^&*()_+');
  });

  test('Task input - show and hide(with enter) input field', () => {
    render(
      <NameInputField index={[0, 0]} typeOfElement="task" taskValue="0 123" />,
    );
    const inputEl = screen.getByLabelText(/add new task/i);
    expect(inputEl).toHaveClass(
      'inputContainer__inputName inputContainer__inputName_hideInput',
    );
    userEvent.click(inputEl);
    expect(inputEl).toHaveClass('inputContainer__inputName');
    fireEvent.keyPress(inputEl, { key: 'Enter', keyCode: 13 });
    expect(inputEl).toHaveClass(
      'inputContainer__inputName inputContainer__inputName_hideInput',
    );
  });

  test('Task input - show and hide(with tab) input field', () => {
    render(
      <NameInputField index={[0, 0]} typeOfElement="task" taskValue="0 123" />,
    );
    const inputEl = screen.getByLabelText(/add new task/i);
    expect(inputEl).toHaveClass(
      'inputContainer__inputName inputContainer__inputName_hideInput',
    );
    userEvent.click(inputEl);
    expect(inputEl).toHaveClass('inputContainer__inputName');
    userEvent.tab();
    expect(inputEl).toHaveClass(
      'inputContainer__inputName inputContainer__inputName_hideInput',
    );
  });

  test('Task input -input some value', () => {
    render(
      <NameInputField index={[0, 0]} typeOfElement="task" taskValue="0 123" />,
    );
    const inputEl = screen.getByLabelText(/add new task/i);
    userEvent.click(inputEl);
    userEvent.type(inputEl, 'some text for input 123!@#$%^&*()_+');
    expect(inputEl).toHaveValue('0 123some text for input 123!@#$%^&*()_+');
    fireEvent.keyPress(inputEl, { key: 'Enter', keyCode: 13 });
    expect(inputEl).toHaveValue('0 123some text for input 123!@#$%^&*()_+');
  });
});
