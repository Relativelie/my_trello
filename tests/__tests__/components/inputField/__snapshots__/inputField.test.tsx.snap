// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input field component snapshot - List Input field component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="inputContainer"
      >
        <input
          aria-label="add new list"
          class="inputContainer__inputName inputContainer__inputName_hideInput"
          data-elemkey="0"
          maxlength="44"
          type="text"
          value=""
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="inputContainer"
    >
      <input
        aria-label="add new list"
        class="inputContainer__inputName inputContainer__inputName_hideInput"
        data-elemkey="0"
        maxlength="44"
        type="text"
        value=""
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Input field component snapshot - Task Input field component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="inputContainer"
      >
        <input
          aria-label="add new task"
          class="inputContainer__inputName inputContainer__inputName_hideInput"
          data-elemkey="0,0"
          maxlength="44"
          type="text"
          value=""
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="inputContainer"
    >
      <input
        aria-label="add new task"
        class="inputContainer__inputName inputContainer__inputName_hideInput"
        data-elemkey="0,0"
        maxlength="44"
        type="text"
        value=""
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
